bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,249 @@
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.etherrain.internal.api;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.SocketTimeoutException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.openhab.binding.etherrain.internal.EtherRainException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EtherRainCommunication} handles communication with EtherRain
+ * controllers so that the API is all in one place
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+public class EtherRainCommunication {
+
+    private static final String BROADCAST_DISCOVERY_MESSAGE = "eviro_id_request1";
+    private static final int BROADCAST_DISCOVERY_PORT = 8089;
+
+    private static final String ETHERRAIN_USERNAME = "admin";
+
+    private static final int HTTP_TIMEOUT = 3;
+
+    private static final int BROADCAST_TIMEOUT = 80;
+
+    private static final String RESPONSE_STATUS_PATTERN = "^\\s*(un|ma|ac|os|cs|rz|ri|rn):\\s*([a-zA-Z0-9\\.]*)(\\s*<br>)?";
+    private static final String BROADCAST_RESPONSE_DISCOVER_PATTERN = "eviro t=(\\S*) n=(\\S*) p=(\\S*) a=(\\S*)";
+
+    private final Logger logger = LoggerFactory.getLogger(EtherRainCommunication.class);
+    private final HttpClient httpClient;
+
+    private final String address;
+    private final int port;
+    private final String password;
+
+    public EtherRainCommunication(String address, int port, String password, HttpClient httpClient) {
+        this.address = address;
+        this.port = port;
+        this.password = password;
+        this.httpClient = httpClient;
+    }
+
+    public static EtherRainUdpResponse autoDiscover() {
+        return updBroadcast();
+    }
+
+    public EtherRainStatusResponse commandStatus() throws EtherRainException, IOException {
+        if (!commandLogin()) {
+            throw new EtherRainException("could not login");
+        }
+
+        List<String> responseList = null;
+
+        responseList = sendGet("result.cgi?xs");
+
+        if (responseList == null) {
+            throw new EtherRainException("Empty Response");
+        }
+
+        Iterator<String> i = responseList.iterator();
```

Comment:
Rather not use iterators those are not common to use (directly) in Java, better use the `for (String line: responseList) {`